src/components/Button/Button.styles.ts

```diff
@@ -41,33 +138,61 @@ export const StyledButtonContainer = styled.button<StyledButtonProps>(({variant,
         },
     };
 
+    // button styles depending upon prop
+    const getButtonStyle = (): ButtonColorMap => {
+        if (destructive) {
+            return destructiveColorMap;
+        } else if (inverted) {
+            return invertedColorMap;
+        }
+        return defaultColorMap;
+    };
+
+    // button text and border color
+    const getButtonColor = (): string => {
+        if (destructive) {
+            return colors.error;
+        } else if (inverted) {
+            return variant === 'primary' ? colors.primary : colors.primaryText;
+        }
+        return colors.primary;
+    };
```

Comment:
instead of executing this function everywhere. Can we execute it here itself and store the value inside the constant. If doing so change the constant name to appropriate name
```suggestion
    const getButtonColor = ((): string => {
        if (destructive) {
            return colors.error;
        } else if (inverted) {
            return variant === 'primary' ? colors.primary : colors.primaryText;
        }
        return colors.primary;
    })();
```

Reply:
change where ever applicable